addons/mail/static/src/models/message/message.js

```diff
@@ -142,26 +114,31 @@ function factory(dependencies) {
                 }
                 data2.originThread = [['insert', originThreadData]];
             }
+            if ('moderation_status' in data) {
+                data2.moderation_status = data.moderation_status;
+            }
             if ('needaction_partner_ids' in data) {
-                const inbox = this.env.messaging.inbox;
-                if (data.needaction_partner_ids.includes(this.env.messaging.currentPartner.id)) {
-                    data2.threadCaches.push(['link', inbox.mainCache]);
-                } else {
-                    data2.threadCaches.push(['unlink', inbox.mainCache]);
-                }
+                data2.isNeedaction = data.needaction_partner_ids.includes(this.env.messaging.currentPartner.id);
             }
             if ('notifications' in data) {
                 data2.notifications = [['insert', data.notifications.map(notificationData =>
                     this.env.models['mail.notification'].convertData(notificationData)
                 )]];
             }
             if ('starred_partner_ids' in data) {
-                const starred = this.env.messaging.starred;
-                if (data.starred_partner_ids.includes(this.env.messaging.currentPartner.id)) {
-                    data2.threadCaches.push(['link', starred.mainCache]);
-                } else {
-                    data2.threadCaches.push(['unlink', starred.mainCache]);
-                }
+                data2.isStarred = data.starred_partner_ids.includes(this.env.messaging.currentPartner.id);
```

Comment:
Side note/out of scope, any reason the formatter returns starred_partner_ids value for any other partner? Otherwise it could be changed to just return `isStarred`.

Reply:
`isStarred`/`isNeedaction`/`isHistory` are current partner-dependent. There is a desire to have formatter agnostic to current partner, so that a single invoke to this formatter can be used to dispatch a same notification to many people.

Reply:
But in this case, it looks like a "private" information, other partners shouldn't need to know what other partners have in history, star, ...

Reply:
> But in this case, it looks like a "private" information, other partners shouldn't need to know what other partners have in history, star, ...

Totally agree with you, it's intimacy leaking and it's bad. I just gave technical motivation to provide all of them instead of current partner specific one, and I don't feel confident to make this change if this results in a huge performance disaster. `message_format` is called a lot and is quite complex as it is, so if you have to call it as many times as there are receivers of a new message, this is a big cost.